zerver/views/custom_profile_fields.py

```diff
@@ -18,17 +19,13 @@
 from zerver.decorator import human_users_only, require_realm_admin
 from zerver.lib.exceptions import JsonableError
 from zerver.lib.external_accounts import validate_external_account_field_data
-from zerver.lib.request import REQ, has_request_variables
 from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import PathOnly, typed_endpoint
 from zerver.lib.types import ProfileDataElementUpdateDict, ProfileFieldData, Validator
 from zerver.lib.users import validate_user_custom_profile_data
 from zerver.lib.validator import (
-    check_bool,
     check_capped_string,
     check_dict,
-    check_dict_only,
-    check_int,
-    check_list,
     check_string,
     check_union,
```

Comment:
So in general we should expect imports from `zerver.lib.validator` to be removed when doing this migration; here there is a bit of remaining code that is using them, but `check_union` is just used in a definition whose only use you deleted. So 

``` diff
diff --git a/zerver/views/custom_profile_fields.py b/zerver/views/custom_profile_fields.py
index a86409a093..e8bd7c09be 100644
--- a/zerver/views/custom_profile_fields.py
+++ b/zerver/views/custom_profile_fields.py
@@ -21,13 +21,10 @@ from zerver.lib.exceptions import JsonableError
 from zerver.lib.external_accounts import validate_external_account_field_data
 from zerver.lib.response import json_success
 from zerver.lib.typed_endpoint import PathOnly, typed_endpoint
-from zerver.lib.types import ProfileDataElementUpdateDict, ProfileFieldData, Validator
+from zerver.lib.types import ProfileDataElementUpdateDict, ProfileFieldData
 from zerver.lib.users import validate_user_custom_profile_data
 from zerver.lib.validator import (
     check_capped_string,
-    check_dict,
-    check_string,
-    check_union,
     validate_select_field_data,
 )
 from zerver.models import CustomProfileField, Realm, UserProfile
@@ -144,11 +141,6 @@ def validate_custom_profile_field_update(
     )
 
 
-check_profile_field_data: Validator[ProfileFieldData] = check_dict(
-    value_validator=check_union([check_dict(value_validator=check_string), check_string])
-)
-
-
 def update_only_display_in_profile_summary(
     existing_field: CustomProfileField,
     requested_field_data: Optional[ProfileFieldData] = None,
```